src/transpile/fusion.hpp

```diff
@@ -149,56 +159,40 @@ void Fusion::optimize_circuit(Circuit& circ,
                               OutputData &data) const {
 
   if (circ.num_qubits < threshold_
-      || !active_
-      || allowed_opset.optypes.find(optype_t::matrix_sequence) == allowed_opset.optypes.end())
+      || !active_)
     return;
 
-  bool ret = false;
-
-  oplist_t optimized_ops;
+  bool applied = false;
 
-  optimized_ops.clear();
-  oplist_t buffer;
-
-  for (const op_t op: circ.ops) {
-    if (can_ignore(op))
+  uint_t fusion_start = 0;
+  for (uint_t op_idx = 0; op_idx < circ.ops.size(); ++op_idx) {
+    if (can_ignore(circ.ops[op_idx]))
       continue;
-    if (!can_apply_fusion(op)) {
-      if (!buffer.empty()) {
-        ret = true;
-        oplist_t optimized_buffer = aggregate(buffer);
-        optimized_ops.insert(optimized_ops.end(), optimized_buffer.begin(), optimized_buffer.end());
-        buffer.clear();
-      }
-      optimized_ops.push_back(op);
-    } else {
-      buffer.push_back(op);
+    if (!can_apply_fusion(circ.ops[op_idx])) {
+      if (fusion_start != op_idx)
```

Comment:
When nesting non-trivial conditionals, it's better if we add curly braces to all the nestings, makes code easier to follow